src/components/operations/operation-details/ko/runtime/graphql-console.ts

```diff
@@ -600,18 +620,60 @@ export class GraphqlConsole {
         this.wsProcessing(true);
         this.displayWsConsole(true);
 
-        let url = this.operationUrl();
-
-        if (url.startsWith("https://")) {
-            url = "wss://" + url.substring(8);
-        } else if (url.startsWith("http://")) {
-            url = "ws://" + url.substring(7)
-        }
+        let url = this.wsUrl();
 
         this.initWebSocket();
         this.ws.connect(url, graphqlSubProtocol);
     }
 
+    private wsUrl(): string {
+        const protocol = this.api().protocols.includes("wss") ? "wss" : "ws";
+        const urlTemplate = this.getRequestPath();
+        const result = `${protocol}://${this.host.hostname()}${Utils.ensureLeadingSlash(urlTemplate)}`;
+        return result;
+    }
+
+    private getRequestPath(getHidden: boolean = false): string {
+        let versionPath = "";
+
+        if (this.api().apiVersionSet && this.api().apiVersion && this.api().apiVersionSet.versioningScheme === "Segment") {
```

Comment:
`this.api()` - gets evaluated multiple time in the same function. I'd suggest doing `const api = this.api()` and use that everywhere below.